core/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -824,15 +823,33 @@ public int countReinitializableVariables(Object entity) {
         return count;
     }
 
-    public boolean isMovable(ScoreDirector<Solution_> scoreDirector, Object entity) {
+    public boolean isMovable(Solution_ workingSolution, Object entity) {
         return isGenuine() &&
-                (effectiveMovableEntitySelectionFilter == null
-                        || effectiveMovableEntitySelectionFilter.accept(scoreDirector, entity));
+                (effectiveMovableEntityFilter == null
+                        || effectiveMovableEntityFilter.test(workingSolution, entity));
     }
 
     @Override
     public String toString() {
         return getClass().getSimpleName() + "(" + entityClass.getName() + ")";
     }
 
+    @FunctionalInterface
```

Comment:
This class appears to be hidden here, and I think it should be moved to a separate package (e.g., ai.timefold.solver.core.impl.heuristic.selector.common.decorator), and it also deserves better documentation.